File: tests/fixtures/typescript/sample-repo/src/falsePositives.ts
// Functions whose names look like effects but whose bodies are pure.
// These exercise the `possible` confidence level only.

export function fetchDefaultColor(): string {
  // expect: Net=possible
  return "blue";
}

export function saveToMemory<T>(items: ReadonlyArray<T>, item: T): T[] {
  // expect: IO=possible, Mut=possible
  // Returns a new array; original `items` is not mutated.
  return [...items, item];
}

export function updateLabel(text: string): string {
  // expect: Mut=possible
  return text.toUpperCase();
}

export function nowStr(): string {
  // expect: Time=possible
  return "now";
}

export function downloadName(_url: string): string {
  // expect: Net=possible
  return "filename.txt";
}

File: tests/fixtures/rust/sample-repo/src/false_positives.rs
// Functions whose names look like effects but whose bodies are pure.
// These exercise the `possible` confidence level only.

pub fn fetch_default_color() -> &'static str {
    // expect: Net=possible
    "blue"
}

pub fn save_to_memory(items: &[&str], item: &str) -> Vec<String> {
    // expect: IO=possible, Mut=possible
    // Returns a NEW Vec; original is not mutated.
    let mut out: Vec<String> = items.iter().map(|s| s.to_string()).collect();
    out.push(item.to_string());
    out
}

pub fn update_label(text: &str) -> String {
    // expect: Mut=possible
    text.to_uppercase()
}

pub fn now_str() -> &'static str {
    // expect: Time=possible
    "now"
}

pub fn download_name(_url: &str) -> &'static str {
    // expect: Net=possible
    "filename.txt"
}
